src/hasSubobject.ts

```diff
@@ -0,0 +1,57 @@
+import { equals } from "./equals";
+import { purry } from "./purry";
+
+/**
+ * Checks if `subObject` is a subobject of `object`, which means for every
+ * property and value in `subObject`, there's the same property in `object`
+ * with an equal value. Equality is checked with `equals`.
+ * @param object the object to test
+ * @param subObject the subobject to test against
+ * @signature
+ *    R.hasSubobject(object, subObject)
+ * @example
+ *    R.hasSubobject({ a: 1, b: 2, c: 3 }, { a: 1, c: 3 }) //=> true
+ *    R.hasSubobject({ a: 1, b: 2, c: 3 }, { b: 4 }) //=> false
+ *    R.hasSubobject({ a: 1, b: 2, c: 3 }, {}) //=> true
+ * @dataFirst
+ * @category Object
+ */
+export function hasSubobject<
```

Comment:
The O in object needs to be capitalized. `subobject` is not a common term so dictionaries like CSpell will warn about this to users.